cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/inmemory/InMemoryStreamAdmin.java

```diff
@@ -35,18 +37,24 @@
  */
 @Singleton
 public class InMemoryStreamAdmin extends InMemoryQueueAdmin implements StreamAdmin {
-
+  private final StreamMetaStore streamMetaStore;
   private final UsageRegistry usageRegistry;
 
   @Inject
-  public InMemoryStreamAdmin(InMemoryQueueService queueService, UsageRegistry usageRegistry) {
+  public InMemoryStreamAdmin(InMemoryQueueService queueService,
+                             UsageRegistry usageRegistry,
+                             StreamMetaStore streamMetaStore) {
     super(queueService);
     this.usageRegistry = usageRegistry;
+    this.streamMetaStore = streamMetaStore;
   }
 
   @Override
   public void dropAllInNamespace(Id.Namespace namespace) throws Exception {
     queueService.resetStreamsWithPrefix(QueueName.prefixForNamedspacedStream(namespace.getId()));
+    for (StreamSpecification spec : streamMetaStore.listStreams(namespace)) {
```

Comment:
Should we instead add a ``removeAll(Id.Namespace)`` method in ``StreamMetastore``, so that it can be made transactional?